src/Listeners/AddApiAttributes.php

```diff
@@ -22,5 +26,36 @@ public function subscribe(Dispatcher $events)
     public function addAdminAttributes(PrepareUnserializedSettings $event)
     {
         $event->settings['flagrow.bazaar.flagrow-host'] = FlagrowApi::getFlagrowHost();
+        $event->settings['flagrow.bazaar.php.memory_limit-met'] = $this->memoryLimitMet();
+        $event->settings['flagrow.bazaar.php.memory_limit'] = ini_get('memory_limit');
+        $event->settings['flagrow.bazaar.php.memory_requested'] = PackageManager::MEMORY_REQUIRED;
+        $event->settings['flagrow.bazaar.file-permissions'] = $this->filePermissions();
+    }
+
+    protected function filePermissions()
+    {
+        $paths = [];
+
+        foreach (['composer.json', 'composer.lock', 'vendor/', null] as $path) {
```

Comment:
We should re-use the logic that already exists in ComposerEnvironment https://github.com/flagrow/bazaar/blob/master/src/Composer/ComposerEnvironment.php#L100

The ComposerEnvironment also checks for the composer home and temporary vendor folder if they exist

We can simply retrieve the ComposerEnvironment out of the IOC and use the checkFilePermissionMethod here. The issue is that this method will throw at the first path exception. We can refactor it to throw a single exception with a list of paths to fix.